### ISSUE TITLE
Hey, there 👋

### ISSUE THREAD

**Issue by Tim-Rambo** [Ames-hub/Rose-Panel#1]
I was wondering if you are still working on this project? 🗬🤔

**Comment by Ames-hub** [comment]
Hey there! it has taken a bit of a sideline as there's been other projects I've been more interested in, and I've hit the roadblock of figuring out how to control the resources assigned to a program (as is a standard feature with most other similar programs) Its not the only problem, but its the hardest one so far.

But yes, I am just as of late, and I'm working on figuring out the roadblock connected to OS compatibility ^^

**Comment by Tim-Rambo** [comment]
Thanks for the update, Amelia!

It's understandable that you've been busy with other projects. Resource allocation and OS compatibility can be quite challenging, so I can see why that's been a roadblock.

Furthermore, I have experience setting up Pterodactyl and have used it many times. I'm currently exploring and wanting to test [Jexactyl](https://github.com/Jexactyl/Jexactyl), which is a Pterodactyl fork.

Furthermore, I'm excited about the potential of this project and believe I can make a meaningful contribution. Let me know how I can get started!

I'm still fairly new to coding, but I'm a fast learner and I have strong problem-solving and other skills that I believe would be valuable to the project. I was looking for a game panel in python, so I can enhance my Python writing skills even more.

**Comment by Ames-hub** [comment]
Yeah, the main problem has been that RosePanel is meant to be able to run on Debian Linux and Windows 10 and above with all features available to both platforms. Resource control is apparently much less easy on Windows.

Very nice :) I've just checked out Jexactyl, looks quite good!

That sounds like a plan, I'll start opening up "issues" for things that are needed in the project. That may be a moment, however. Since I am currently in the process of updating this project from py 3.11 to 3.12 and considering making use of frameworks such as django for the webserver. I hope this project can help improve your skills in Python ^^

**Comment by Tim-Rambo** [comment]
> frameworks such as django for the webserver.

Have you heard of reflex framework? [Reflex website](https://reflex.dev/) it's a full stack web framework in pure python. Could be interesting to use with the rose panel, maybe?

---

> Resource control is apparently much less easy on Windows.

[Pypi package link](https://pypi.org/project/psutil/) I see that there is a panel called [Ajenti](https://github.com/ajenti/ajenti) that uses this package, which is a Cross-platform lib for process and system monitoring in Python.

---

> That sounds like a plan, I'll start opening up "issues" for things that are needed in the project.

Thank you, this will help make me understand more of the project and what needs to be worked on.

---

> I hope this project can help improve your skills in Python ^^

Thank you for letting me force myself out of my comfort zone to really get hands on with coding, with having ADHD I have been struggling on how to start etc. I feel like this will be a kick in the ass to force myself into a project while also getting some help. Thank you, for the opportunity.

---

Question: Are you planning to use docker for the game server, like pterodactyl does for wings + game containers, or do you want everything under one?

---

If there is anything that needs to be researched, let me know.


---

Forgot to mention I got some experience in setting up linux server's + I run GarudaOS which is Arch based.

**Comment by Ames-hub** [comment]
> Have you heard of reflex framework? [Reflex website](https://reflex.dev/) it's a full stack web framework in pure python. Could be interesting to use with the rose panel, maybe?

I'm going to test run it in a local project just to see how it goes. The major reason I intend to go with Django, or hell, even just an nginx docker container, is because I am familiar with HTML, CSS and JS. But we'll see how it goes depending on if it doesn't prove to be too much of a hurdle. Thanks for the recommendation ^^

---

> [Pypi package link](https://pypi.org/project/psutil/) I see that there is a panel called [Ajenti](https://github.com/ajenti/ajenti) that uses this package, which is a Cross-platform lib for process and system monitoring in Python.

Hey, yes that's true. Psutil can be used for process and system monitoring, however, it cannot be used to control. On Linux, there are such things as C-Groups which do what I need, but window's lacks that functionality. If Ajenti controls process resources, its likely since they're a linux only program and use C-Groups.

Recently, I've discovered a windows program called Process Lasso which I will likely end up using for windows compatibility and c-groups for Linux compatibility. We'll see how it develops though.

---

> Thank you for letting me force myself out of my comfort zone to really get hands on with coding, with having ADHD I have been struggling on how to start etc. I feel like this will be a kick in the ass to force myself into a project while also getting some help. Thank you, for the opportunity.

Its not a problem! If you like, you can add me on discord: "friendlyfox.exe" since talking on Github can be, awkward to say the least. Its fine if you'd rather not however. But I'm always open to help with projects :) Especially difficult ones, such as this.

---

> Question: Are you planning to use docker for the game server, like pterodactyl does for wings + game containers, or do you want everything under one?

I'm intending on using a system like Wings for Pterodactyl, you may notice the file "Thorns.py", my badly failed attempt at a recreation of it.

Basically, Rosepanel was intended to be a panel with the simplicity of Pufferpanel but the features of Pterodactyl panel. Which is a hard task, as both are quite amazing programs. But with enough time, I reckon it can be done.

But generally, I intend to make RosePanel as much of a "All in one" program as I can, as its meant to be as simple to use AND install.

---

> Forgot to mention I got some experience in setting up linux server's + I run GarudaOS which is Arch based.

Good to know ^^

**Comment by Tim-Rambo** [comment]
> > Have you heard of reflex framework? [Reflex website](https://reflex.dev/) it's a full stack web framework in pure python. Could be interesting to use with the rose panel, maybe?
> 
> I'm going to test run it in a local project just to see how it goes. The major reason I intend to go with Django, or hell, even just an nginx docker container, is because I am familiar with HTML, CSS and JS. But we'll see how it goes depending on if it doesn't prove to be too much of a hurdle. Thanks for the recommendation ^^

I see, I played around with reflex a while back, I got to say It's pretty neat. I wanted to do a project with it, but never knew where to begin with it, since coding/programming can be pretty daunting.

---



> > [Pypi package link](https://pypi.org/project/psutil/) I see that there is a panel called [Ajenti](https://github.com/ajenti/ajenti) that uses this package, which is a Cross-platform lib for process and system monitoring in Python.
> 
> Hey, yes that's true. Psutil can be used for process and system monitoring, however, it cannot be used to control. On Linux, there are such things as C-Groups which do what I need, but window's lacks that functionality. If Ajenti controls process resources, its likely since they're a linux only program and use C-Groups.
> 
> Recently, I've discovered a windows program called Process Lasso which I will likely end up using for windows compatibility and c-groups for Linux compatibility. We'll see how it develops though.

Right, silly me, I forgot you are doing windows too. With that said, wouldn't one be able to fully control the game resources trough docker containers, like pterodactyl is doing? Or are you thinking more about the panel resources?

---



> > Thank you for letting me force myself out of my comfort zone to really get hands on with coding, with having ADHD I have been struggling on how to start etc. I feel like this will be a kick in the ass to force myself into a project while also getting some help. Thank you, for the opportunity.
> 
> Its not a problem! If you like, you can add me on discord: "friendlyfox.exe" since talking on Github can be, awkward to say the least. Its fine if you'd rather not however. But I'm always open to help with projects :) Especially difficult ones, such as this.

I've added you :) 

---


> > Question: Are you planning to use docker for the game server, like pterodactyl does for wings + game containers, or do you want everything under one?
> 
> I'm intending on using a system like Wings for Pterodactyl, you may notice the file "Thorns.py", my badly failed attempt at a recreation of it.
> 
> Basically, Rosepanel was intended to be a panel with the simplicity of Pufferpanel but the features of Pterodactyl panel. Which is a hard task, as both are quite amazing programs. But with enough time, I reckon it can be done.
> 
> But generally, I intend to make RosePanel as much of a "All in one" program as I can, as its meant to be as simple to use AND install.

If RosePanel was ever to be hard to install and setup, one could provide an official .bat and .shell installation script to make it simple like pufferpanel but has the features of pterodactyl + more. But again going for that all in one solution would make it easier for the end user to be able to probably do 1 command for install and rest is just setting up and configuring the panel in a simple GUI of some sort.

There are unofficial ones for pterodactyl, but one does not get support if you use one. RosePanel should help the end user there.

My friend told me about this KISS, **K**eep **I**t **S**tupid **S**imple

---


> > Forgot to mention I got some experience in setting up linux server's + I run GarudaOS which is Arch based.
> 
> Good to know ^^

I have a ton of experience in the backend of things when it comes to servers, game servers and so on. Currently setting up an ubuntu 24.04 GNU/Linux server. But no coding only problem-solving and finding solution myself instead of asking someone all the time, and have to wait for hours for an answer to a simple question. With AI, it has become easier for me to dabble into coding which I find nice, but I do not like the No Code direction everyone is going in and with it replacing us. lol, it was nice having something to ask for direction ideas etc. It has also given me a push out of the comfort zone.

For AI, I use Deepseek + [Aider](https://aider.chat/) It has come so far it is writing its own updates.

**Comment by Ames-hub** [comment]
> Right, silly me, I forgot you are doing windows too. With that said, wouldn't one be able to fully control the game resources trough docker containers, like pterodactyl is doing? Or are you thinking more about the panel resources?

I don't actually know. I was mainly thinking of controlling the game servers that Rose panel runs, but I don't actually know much about how docker works with regard to resource control. Something to look into

---

> There are unofficial ones for pterodactyl, but one does not get support if you use one. RosePanel should help the end user there.

Yeah definitely. Personally I want to avoid it and just have it be something like "Run the python file and the app will auto handle whatever it needs" 

---

> My friend told me about this KISS, Keep It Stupid Simple

This fr. I wish I followed that rule 6 months ago when I had been actively working on this program lmfao

---

> I have a ton of experience in the backend of things when it comes to servers, game servers and so on. Currently setting up an ubuntu 24.04 GNU/Linux server. But no coding only problem-solving and finding solution myself instead of asking someone all the time, and have to wait for hours for an answer to a simple question. With AI, it has become easier for me to dabble into coding which I find nice, but I do not like the No Code direction everyone is going in and with it replacing us. lol, it was nice having something to ask for direction ideas etc. It has also given me a push out of the comfort zone.

That's great, it'll come in handy for when you're building apps. But yeah totally, I probably wouldn't have gotten into programming if it wasn't for AI. I had a lot of questions and things I needed explained to me that I didn't want to sit through hour long video's for.

---

> I've added you :)

Lovely! I'll close this issue, and we can talk there.
